src/Effect/Aff.js

```diff
@@ -320,24 +320,32 @@ var Aff = function () {
 
           case ASYNC:
             status = PENDING;
-            step   = runAsync(util.left, step._1, function (result) {
-              return function () {
-                if (runTick !== localRunTick) {
-                  return;
-                }
-                runTick++;
-                Scheduler.enqueue(function () {
-                  // It's possible to interrupt the fiber between enqueuing and
-                  // resuming, so we need to check that the runTick is still
-                  // valid.
+            var asyncAction = step._1;
```

Comment:
The `tmp` register is for stuff like this. I'd prefer to not introduce a `var` in the middle of a switch.